openff/recharge/tests/conformers/test_conformers.py

```diff
@@ -40,7 +40,7 @@ def test_generate_omega_conformers_error():
         ConformerGenerationError, match="ELF10 conformer selection failed"
     ):
 
-        ConformerGenerator.generate(Molecule.from_smiles("[Mg]"), ConformerSettings())
```

Comment:
This fails with 0.11.1 because it's detected as a radical but passes with 0.11.0.

I don't care at all how this is parsed - the point of this test is to feed a conformer that will fail Omega conformer generation. Any ideas of a small molecule that reliably does this but is not subject to opinionated parsing issues?

cc: @j-wags @lilyminium

Reply:
Would something like `[Fe][Fe]` work?

Reply:
Yes, except I asked the question poorly. The molecule needs to pass conformer generation but fail ELF10 conformer selection. I don't recalled the detailed differences between the two off the top of my head, but the traceback and relevant code is below.

Keep in mind this is something that's useful for ensure the tests don't lose coverage, not something that's a part of the source code



https://github.com/openforcefield/openff-recharge/blob/7d03bd61f41666257d703e469de410de197894b9/openff/recharge/conformers/_conformers.py#L62-L75

```
pytest openff/recharge/tests/conformers/test_conformers.py --lf
=============================================== test session starts ===============================================
platform darwin -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/mattthompson/software/openff-recharge
plugins: xdist-2.5.0, forked-1.4.0, cov-4.0.0
collected 11 items / 10 deselected / 1 selected
run-last-failure: rerun previous 1 failure

openff/recharge/tests/conformers/test_conformers.py F                                                       [100%]

==================================================== FAILURES =====================================================
______________________________________ test_generate_omega_conformers_error _______________________________________

    def test_generate_omega_conformers_error():

        with pytest.raises(
            ConformerGenerationError, match="Failed to generate conformers using OMEGA"
        ):

            ConformerGenerator.generate(
                Molecule.from_smiles("C(S(=O)[O-])(Cl)(Cl)Cl", allow_undefined_stereo=True),
                ConformerSettings(method="omega", sampling_mode="sparse", max_conformers=1),
            )

        with pytest.raises(
            ConformerGenerationError, match="ELF10 conformer selection failed"
        ):

>           ConformerGenerator.generate(Molecule.from_smiles("[Fe][Fe]"), ConformerSettings())

openff/recharge/tests/conformers/test_conformers.py:43:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'openff.recharge.conformers._conformers.ConformerGenerator'>
molecule = Molecule with name '' and SMILES '[Fe][Fe]'
settings = ConformerSettings(method='omega-elf10', sampling_mode='dense', max_conformers=5)

    @classmethod
    def generate(
        cls,
        molecule: "Molecule",
        settings: ConformerSettings,
    ) -> List[unit.Quantity]:
        """Generates a set of conformers for a given molecule.

        Notes
        -----
        * All operations are performed on a copy of the original molecule so that it
        will not be mutated by this function.

        Parameters
        ----------
        molecule
            The molecule to generate conformers for.
        settings
            The settings to generate the conformers according to.
        """

        if "omega" in settings.method:
>           conformers = cls._generate_omega_conformers(molecule, settings)

openff/recharge/conformers/_conformers.py:111:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'openff.recharge.conformers._conformers.ConformerGenerator'>
molecule = Molecule with name '' and SMILES '[Fe][Fe]'
settings = ConformerSettings(method='omega-elf10', sampling_mode='dense', max_conformers=5)

    @classmethod
    def _generate_omega_conformers(
        cls,
        molecule: "Molecule",
        settings: ConformerSettings,
    ) -> List[unit.Quantity]:

        oe_molecule = molecule.to_openeye()

        from openeye import oeomega, oequacpac

        if settings.sampling_mode == "sparse":
            omega_options = oeomega.OEOmegaOptions(oeomega.OEOmegaSampling_Sparse)
        elif settings.sampling_mode == "dense":
            omega_options = oeomega.OEOmegaOptions(oeomega.OEOmegaSampling_Dense)
        else:
            raise NotImplementedError()

        omega = oeomega.OEOmega(omega_options)
        omega.SetIncludeInput(False)
        omega.SetCanonOrder(False)

        if not omega(oe_molecule):
>           raise ConformerGenerationError("Failed to generate conformers using OMEGA")
E           openff.recharge.conformers.exceptions.ConformerGenerationError: Failed to generate conformers using OMEGA

openff/recharge/conformers/_conformers.py:62: ConformerGenerationError

During handling of the above exception, another exception occurred:

    def test_generate_omega_conformers_error():

        with pytest.raises(
            ConformerGenerationError, match="Failed to generate conformers using OMEGA"
        ):

            ConformerGenerator.generate(
                Molecule.from_smiles("C(S(=O)[O-])(Cl)(Cl)Cl", allow_undefined_stereo=True),
                ConformerSettings(method="omega", sampling_mode="sparse", max_conformers=1),
            )

        with pytest.raises(
            ConformerGenerationError, match="ELF10 conformer selection failed"
        ):

>           ConformerGenerator.generate(Molecule.from_smiles("[Fe][Fe]"), ConformerSettings())
E           AssertionError: Regex pattern 'ELF10 conformer selection failed' does not match 'Failed to generate conformers using OMEGA'.

openff/recharge/tests/conformers/test_conformers.py:43: AssertionError
---------------------------------------------- Captured stderr call -----------------------------------------------
Warning: : Failed due to unspecified stereochemistry
Warning: OEMMFFParams::PrepMol() : unable to type atom 0 Fe on residue UNL-1
Warning: OEMMFFParams::PrepMol() : unable to type atom 1 Fe on residue UNL-1
Warning: : Force field setup failed due to missing parameters
------------------------------------------------ Captured log call ------------------------------------------------
WARNING  openff.toolkit.utils.openeye_wrapper:openeye_wrapper.py:1126 Warning (not error because allow_undefined_stereo=True): OEMol has unspecified stereochemistry. oemol.GetTitle():
Problematic atoms are:
Atom atomic num: 16, name: , idx: 1, aromatic: False, chiral: True with bonds:
bond order: 1, chiral: False to atom atomic num: 6, name: , idx: 0, aromatic: False, chiral: False
bond order: 2, chiral: False to atom atomic num: 8, name: , idx: 2, aromatic: False, chiral: False
bond order: 1, chiral: False to atom atomic num: 8, name: , idx: 3, aromatic: False, chiral: False
============================================= short test summary info =============================================
FAILED openff/recharge/tests/conformers/test_conformers.py::test_generate_omega_conformers_error - AssertionErro...
======================================== 1 failed, 10 deselected in 1.34s =========================================